services/121-service/src/programs/program/program.service.ts

```diff
@@ -500,30 +503,20 @@ export class ProgramService {
       throw new HttpException({ errors }, 404);
     }
 
-    const connections = await this.connectionRepository
-      .createQueryBuilder('table')
-      .where('1 =1')
-      .getMany();
+    const includedConnections = await this.getIncludedConnections(programId);
 
-    const includedConnections = [];
-    for (let connection of connections) {
-      if (connection.programsIncluded.includes(programId)) {
-        includedConnections.push(connection);
-      }
-    }
-    if (includedConnections.length < 1){
+    if (includedConnections.length < 1) {
       const errors = 'There are no included PA for this program';
       throw new HttpException({ errors }, 404);
     }
 
-    const availableFunds = await this.fundingService.getProgramFunds(programId)
-    const fundsNeeded = amount * includedConnections.length
+    const availableFunds = await this.fundingService.getProgramFunds(programId);
+    const fundsNeeded = amount * includedConnections.length;
     if (fundsNeeded > availableFunds) {
       const errors = 'Not enough available funds';
       throw new HttpException({ errors }, 404);
```

Comment:
This could be an error, but 404 doesn't sound the best fit...

Because technically the request would be well-formed and successfully transmitted; But the outcome of the calculation is just "not valid" or "not good enough".

So 200, with a `{ status: 'error', message: 'Insufficient funds' }`; Or similar...

Reply:
Created: https://github.com/global-121/121-platform/compare/master...RubenGeo:statusPayout?expand=1